### ISSUE TITLE
Batch mode problem using nn.SpatialClassifier

### ISSUE THREAD

**Issue by jblan016** [torch/torch7#709]
Hi, I trained a model using minibatches of size 128. With 20x20 grayscale images.
Here is the model that I've successfully trained with a locally connected convolutional layer as in (the N1 layer of http://liris.cnrs.fr/Documents/Liris-6132.pdf):
model = nn.Sequential()
c = nn.Concat(2)
model:add(nn.SpatialConvolutionMM(1,32,3,3,1,1)) --I(1x20x20)->O(32x18x18)
--model:add(nn.Dropout())
model:add(nn.SpatialMaxPooling(3,3,2,2,1,1)) --O(32x9x9)
model:add(nn.ReLU())
model:add(nn.SpatialConvolutionMM(32,32,3,3,1,1))--O(32x7x7)
--model:add(nn.Dropout())
model:add(nn.ReLU())
model:add(nn.SpatialMaxPooling(3,3,2,2,0,0)) --O(32x3x3)
--model:add(nn.SpatialConvolutionMM(32,32,3,3,1,1))--O(32x1x1)
for i=1,32 do
  local t = nn.Sequential()
  t:add(nn.Select(2,i))
  t:add(nn.Unsqueeze(2))
  t:add(nn.SpatialConvolutionMM(1,1,3,3,1,1))
  t:add(nn.ReLU())
  c:add(t)
end
model:add(c)
  model:add(nn.Reshape(32))
model:add(nn.Linear(32,2))
model:add(nn.LogSoftMax())

In another script, I split the classifier and the CNN with :
...
network_temp = torch.load(model_path)
model = nn.Sequential()

for i=1,7 do
    model:add(network_temp.modules[i])
end
classifier1 = nn.Sequential()
for i=8,10 do 
    classifier1:add(network_temp.modules[i])
end

classifier = nn.SpatialClassifier(classifier1)
model:add(classifier)
model:cuda()
...

Then what I do is I create an image pyramid with:
...
packer = nn.PyramidPacker(model, scales)  
    unpacker = nn.PyramidUnPacker(model)
...
pyramid, coordinates = packer:forward(im)  --Here im is a 1x400x400 for example
...
So far so good no error but when I try to create a probability heat map with:
det = model:forward(pyramid:cuda())

I get the following error:
qlua: /usr/local/share/lua/5.1/nn/Container.lua:69: 
In 7 module of nn.Sequential:
In 1 module of nn.Concat:
In 2 module of nn.Sequential:
/usr/local/share/lua/5.1/nn/Unsqueeze.lua:43: input feature map dim (numInputDims) must be <= input:dim()

I used multiple versions of the model that did compile using table layers and specifying the number of inputs to 3. The training is always fine but when I apply the model as a filter using NO BATCH MODE it stops working. If I remove the locally connected convolutional layer then everything is fine, and I get a heat map. Any tips/pointers for making my model work as a filter?

**Comment by jblan016** [comment]
I fixed it using replacing my model with:
model = nn.Sequential()
model:add(nn.SpatialConvolutionMM(1,32,3,3,1,1)) --I(bx1x20x20)->O(bx32x18x18)
model:add(nn.SpatialMaxPooling(3,3,2,2,1,1)) --O(bx32x9x9)
model:add(nn.ReLU())
model:add(nn.SpatialConvolutionMM(32,32,3,3,1,1))--O(bx32x7x7)
model:add(nn.ReLU())
model:add(nn.SpatialMaxPooling(3,3,2,2,1,1)) --O(bx32x4x4)
model:add(nn.SpatialConvolutionMap(nn.tables.oneToOne(32),4,4,1,1))
model:add(nn.ReLU())
model:add(nn.Reshape(32))-- O(bx32)
model:add(nn.Linear(32,2)) --O(bx2)
model:add(nn.LogSoftMax())

Yep thank god the documentation is clear as frech mineral water.
